lib/request.js

```diff
@@ -50,9 +69,15 @@ class MockSocket extends EventEmitter {
  * @param {Object} [options.cookies]
  * @param {Object} [options.headers]
  * @param {Object} [options.query]
+ * @param {Object} [options.customRequestType]
  * @param {any} [options.payload]
  */
 function Request (options) {
+  if (options.customRequestType) {
+    assert(options.customRequestType.prototype !== undefined)
+    return new MakeCustomRequest(this, options)
+  }
```

Comment:
Can you move this if to where Request is instantiated?

Reply:
My idea was to keep it transparent to the `lib/request` clients. But sure, will do